recastjs/src/recastjs.cpp

```diff
@@ -0,0 +1,577 @@
+#include "recastjs.h"
+#include "Recast.h"
+#include "DetourNavMesh.h"
+#include "DetourNavMeshBuilder.h"
+#include "DetourNavMesh.h"
+#include "DetourNavMeshQuery.h"
+
+#include <stdio.h>
+#include <vector>
+#include <float.h>
+#include <algorithm>
+#include <math.h>
+#include <sstream>
+#include <iostream>
+
+void Log(const char* str)
+{
+    std::cout << std::string(str) << std::endl;
+}
+
+int g_seed = 1337;
+inline int fastrand() 
+{ 
+    g_seed = (214013*g_seed+2531011); 
+    return (g_seed>>16)&0x7FFF; 
+} 
+
+inline float r01()
+{
+    return ((float)fastrand())*(1.f/32767.f);
+}
+
+void NavMesh::destroy()
+{
+    dtFreeNavMesh(m_navMesh);
+    dtFreeNavMeshQuery(m_navQuery);
+}
+
+void NavMesh::build(const float* positions, const int positionCount, const int* indices, const int indexCount, const rcConfig& config)
+{
```

Comment:
there are a bunch of mixed conventions in this function. some variables are prefixed with `m`, some with `m_` but none of them are member variables. is that intentional?

Reply:
true! fixed.